src/libYARP_dev/include/yarp/dev/ILocalization2D.h

```diff
@@ -32,6 +40,18 @@ class yarp::dev::ILocalization2D
      */
     virtual ~ILocalization2D() {}
 
+    /**
+    * Gets the current status of the localization task.
+    * @return true/false
+    */
+    virtual bool   getLocalizationStatus(LocalizationStatusEnum& status) = 0;
+
+    /**
+    * Gets a set of pose estimates computed by the localization algorithm.
```

Comment:
What is the relation between this set of pose estimates and the position returned by the `getCurrentPosition` method?

Reply:
This method return all the possibile poses evaluated by the particle filter (the internal status of the algorithm)
getCurrentPosition returns the pose with highest probability. ROS returns also the covariance (currently skipped).